### ISSUE TITLE
Convert project to Kotlin

### ISSUE THREAD

**Issue by erluxman** [iSoron/uhabits#300]
Are there any plans about migrating this project to kotlin ?

**Comment by iSoron** [comment]
Yes, I guess we will eventually migrate to Kotlin, but this is not a high priority right now.

**Comment by erluxman** [comment]
Are Kotlin PRs welcomed ?

**Comment by iSoron** [comment]
@erluxman Sure, Kotlin PRs are very welcome. I have started converting the `uhabits-android` module to Kotlin a few days ago, on the branch `feature/kotlin`. I would like to have that module completely converted first, before we start converting the others.

**Comment by erluxman** [comment]
That is amazing..

**Comment by alex-schwartzman** [comment]
Hi, @iSoron 

I'm afraid, that big-bang approach of migrating to Kotlin did not succeed. Should we now switch to another approach where developers will be welcome to sporadically rewrite java files into Kotlin here and there?  

I'd love to contribute if we go this way.

Thanks.
Alex.

**Comment by aschrijver** [comment]
While Kotlin is gaining a lot of traction these days, given that it still has a much smaller developer base than Java, will a migration not unnecessarily increase the barrier to contribute to this project by a significant amount?

**Comment by olegivo** [comment]
Nowadays the Kotlin is one of standard Android languages. I assume the migration is useful especially in light of the multi-platform development direction.
I would like to contribute of step by step migrations.

**Comment by iSoron** [comment]
For anyone working on this, please do not submit changes to any files in `uhabits/android/uhabits-core`. All these files will be replaced by `uhabits/core`, which is a rewrite in Kotlin Multiplatform. Changes to `uhabits/android/uhabits-android` are still welcome.

**Comment by iSoron** [comment]
Closing this in favor of #685. Pull requests are very welcome.
